src/coreclr/vm/readytoruninfo.cpp

```diff
@@ -1771,5 +1789,161 @@ ModuleBase* CreateNativeManifestModule(LoaderAllocator* pLoaderAllocator, IMDInt
+    }
+    return !!((uint8_t)typeGenericInfo & (uint8_t)ReadyToRunTypeGenericInfo::GenericCountMask);
+}
+
+HRESULT ReadyToRun_TypeGenericInfoMap::IsGenericNoThrow(mdTypeDef input, bool *pIsGeneric, IMDInternalImport* pImport) const
+{
+    bool foundResult;
+    bool result;
+    HRESULT hr;
+    ReadyToRunTypeGenericInfo typeGenericInfo = GetTypeGenericInfo(input, &foundResult);
+    if (!foundResult)
+    {
+        HENUMInternalHolder hEnumTyPars(pImport);
+        IfFailRet(hEnumTyPars.EnumInitNoThrow(mdtGenericParam, input));
+        result = (pImport->EnumGetCount(&hEnumTyPars) != 0);
+    }
+    else
+        result = !!((uint8_t)typeGenericInfo & (uint8_t)ReadyToRunTypeGenericInfo::GenericCountMask);
+
+    *pIsGeneric = result;
+    return S_OK;
+}
+
+uint32_t ReadyToRun_TypeGenericInfoMap::GetGenericArgumentCount(mdTypeDef input, IMDInternalImport* pImport) const
+{
+    bool foundResult;
+    ReadyToRunTypeGenericInfo typeGenericInfo = GetTypeGenericInfo(input, &foundResult);
+    uint32_t count = ((uint8_t)typeGenericInfo & (uint8_t)ReadyToRunTypeGenericInfo::GenericCountMask);
+    if (count > 2)
+        foundResult = false;
+    
+    if (!foundResult)
+    {
+        HENUMInternalHolder hEnumTyPars(pImport);
+        hEnumTyPars.EnumInit(mdtGenericParam, input);
+        return pImport->EnumGetCount(&hEnumTyPars);
+    }
+    return count;
+}
+
+HRESULT ReadyToRun_TypeGenericInfoMap::GetGenericArgumentCountNoThrow(mdTypeDef input, uint32_t *pCount, IMDInternalImport* pImport) const
+{
+    HRESULT hr;
+    bool foundResult;
+    ReadyToRunTypeGenericInfo typeGenericInfo = GetTypeGenericInfo(input, &foundResult);
+    uint32_t count = ((uint8_t)typeGenericInfo & (uint8_t)ReadyToRunTypeGenericInfo::GenericCountMask);
+    if (count > 2)
+        foundResult = false;
+    
+    if (!foundResult)
+    {
+        HENUMInternalHolder hEnumTyPars(pImport);
+        IfFailRet(hEnumTyPars.EnumInitNoThrow(mdtGenericParam, input));
+        count = pImport->EnumGetCount(&hEnumTyPars);
+    }
+
+    hr = S_OK;
+    *pCount = count;
+    return hr;
+}
+
+bool ReadyToRun_TypeGenericInfoMap::HasVariance(mdTypeDef input, bool *foundResult) const
+{
+    ReadyToRunTypeGenericInfo typeGenericInfo = GetTypeGenericInfo(input, foundResult);
+    return !!((uint8_t)typeGenericInfo & (uint8_t)ReadyToRunTypeGenericInfo::HasVariance);
+}
+
+bool ReadyToRun_TypeGenericInfoMap::HasConstraints(mdTypeDef input, bool *foundResult) const
+{
+    ReadyToRunTypeGenericInfo typeGenericInfo = GetTypeGenericInfo(input, foundResult);
+    return !!((uint8_t)typeGenericInfo & (uint8_t)ReadyToRunTypeGenericInfo::HasVariance);
+}
+
+bool ReadyToRun_MethodIsGenericMap::IsGeneric(mdMethodDef input, bool *foundResult) const
+{
+    uint32_t rid = RidFromToken(input);
+    if ((rid > MethodCount) || (rid == 0))
+    {
+        *foundResult = false;
+        return 0;
+    }
+
+    uint8_t chunk = ((uint8_t*)&MethodCount)[((rid - 1) / 8) + 4];
```

Comment:
Nit - if you want to be consistent with your implementation of the other tables (and possibly also make the code easier to understand), you might want to consider replacing 4 with <code>sizeof(uint32_t)</code> to make it obvious it represents the initial count entry in the table.